server/encryption/localkey_handler.go

```diff
@@ -0,0 +1,217 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+	"os"
+
+	"github.com/google/tink/go/kwp/subtle"
+)
+
+const (
+	// DataKeyLength provides the length for data key in bytes.
+	// It is recommended to use an authentication key with 32 or 64 bytes.
+	// The data key length must be either
+	// 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256 modes.
+	DataKeyLength int = 32
+)
+
+var (
+	masterKeyVarName = "LOCAL_MASTER_KEY"
+)
+
+// LocalEncryptionHandler provides functionalities to load secret key
+// from environment variables
+type LocalEncryptionHandler struct {
+	defaultDKS  []byte
+	keywrapper  *subtle.KWP
+	blockCypher *cipher.AEAD
+}
+
+// NewLocalEncriptionHandler generates a new instance of LocalEncryptionHandler
+func NewLocalEncriptionHandler() (*LocalEncryptionHandler, error) {
+	localEncryptionHandler := LocalEncryptionHandler{}
+
+	// Init key wrapper
+
+	masterKeyStr := os.Getenv(masterKeyVarName)
+	masterKey := []byte(masterKeyStr)
+	kwp, err := subtle.NewKWP(masterKey)
+
+	if err != nil {
+		return nil, err
+	}
+	localEncryptionHandler.keywrapper = kwp
+
+	return &localEncryptionHandler, nil
+}
+
+// generateDKS retrieves the pre-configurated encryption key
+// from the environment variables.
+func (handler *LocalEncryptionHandler) generateDKS() ([]byte, error) {
+	key := make([]byte, DataKeyLength)
+
+	_, err := rand.Read(key)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return key, nil
+
+}
+
+func (handler *LocalEncryptionHandler) wrapDKS(dks []byte) ([]byte, error) {
+	// use Tinker to wrap Data Key
+	// https://github.com/google/tink/commit/22467ef7273d73b2d65e4b50310aab4af006bb7e
+	wrappededKey, err := handler.keywrapper.Wrap(dks)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return wrappededKey, nil
+
+}
+
+func (handler *LocalEncryptionHandler) unwrapDKS(wrappedDKS []byte) ([]byte, error) {
+	// use Tinker to unwrap Data Key
+	// https://github.com/google/tink/commit/22467ef7273d73b2d65e4b50310aab4af006bb7e
+	key, err := handler.keywrapper.Unwrap(wrappedDKS)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return key, nil
+}
+
+func (handler *LocalEncryptionHandler) encryptData(dks []byte, plaintextData []byte) ([]byte, error) {
+	// Init cypher in GCM
+
+	block, err := aes.NewCipher(dks)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCM(block)
+
+	if err != nil {
+		return nil, err
+	}
+	// init nonce
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		panic(err.Error())
```

Comment:
Why panic?

Reply:
mistake I think. a `return nil, error` is good enought in this case IMO.